lib/libc/arcmwdt/threading.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_MULTITHREADING
+
+#include <init.h>
+#include <kernel.h>
+#include <sys/__assert.h>
+#include <sys/mutex.h>
+#include <logging/log.h>
+#include <../lib/src/c/inc/internal/thread.h>
+
+#ifndef CONFIG_USERSPACE
+#define ARCMWDT_DYN_LOCK_SZ	(sizeof(struct k_mutex))
+#define ARCMWDT_MAX_DYN_LOCKS	10
```

Comment:
Should this be a Kconfig config?

Reply:
It could, but the value is more tied to the MetaWare C library and its use of mutexes to protect things like file handles, than to tied Zephyr.

Reply:
This locks are used internally in libraries, so their number don't expected to be changed / configured by user itself.
That's why we decided to avoid moving it to Kconfig (at least for now).

Reply:
Fair enough.

Users could potentially set this to a lower value in order to save memory, given that they do not call any MW libc functions that require these many locks, but that would be a very tricky thing to estimate and configure in practice.